samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -104,14 +116,29 @@ public synchronized void stop() {
       coordinatorListener.onJobModelExpired();
     }
 
+    metrics.isLeader.set(0);
     debounceTimer.stopScheduler();
     zkController.stop();
 
+    shutdownMetrics();
     if (coordinatorListener != null) {
       coordinatorListener.onCoordinatorStop();
     }
   }
 
+  public void startMetrics() {
+    for (MetricsReporter reporter:reporters.values()) {
+      reporter.register("samza-container-" + processorId, metrics.getMetricsRegistry());
```

Comment:
s/samza-container-/job-coordinator-/

This parameter represent the source of this set of metrics. That is the jobcoordinator component of the processor.